File: Week6/ProblemSet6/sentimental-mario-more/mario.py
# TODO

def main():
    # Check if user inputted the correct values
    height = input("Height: ")
    height = height_check(height)
    while (height == 0):
        height = input("Height: ")
        height = height_check(height)

    n_row = 1
    while (n_row <= height):
        print_spaces(height, n_row)  #
        print_hashtag(n_row)  # Draw the left side
        print("  ", end="")  # Space in between to separate
        print_hashtag(n_row)  # Draw the right side
        print()  # Then moves on to next line
        n_row += 1


def print_spaces(height, n_row):
    column = n_row
    while (column < height):
        print(" ", end="")  # Print spaces
        column += 1


def print_hashtag(n_row):
    column = 1
    while (column <= n_row):
        print("#", end="")
        column += 1


def height_check(user_input):
    # Only accept 1 - 8 numbers and numbers only
    try:
        user_input = int(user_input)
        if (user_input >= 1) and (user_input <= 8):
            return user_input
        pass
    except ValueError:
        pass
    return 0  # The input didn't violated any rules


if __name__ == "__main__":
    main()

File: Week1/Mario/mario-more/mario.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>


void print_spaces(int height, int n_row)
{
    for (int column = n_row; column < height; column++)
    {
        printf(" ");  // Print Spaces
    }
}

void print_hashtag(int height, int n_row)
{
    for (int column = 1; column <= n_row; column++)
    {
           printf("#");
    }
}

int height_check(int user_input)
{
    // Only accept 1 - 8 numbers and numbers only
    if ((user_input < 1) || (user_input > 8))
    {
        return 1; // The user_input inputted negative and outside of 8 numbers
    }
    return 0;  // The input didn't violated any rules
}

int main(void)
{
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height_check(height));

    for (int row = 1; row <= height; row++)
    {
        print_spaces(height, row);
        print_hashtag(height, row);
        printf("  ");  // Space in between
        print_hashtag(height, row);
        printf("\n");
    }
}

